src/theory/sets/theory_sets_private.cpp

```diff
@@ -1145,7 +1145,7 @@ bool TheorySetsPrivate::collectModelValues(TheoryModel* m,
         {
           for (const std::pair<const Node, Node>& itmm : emems)
           {
-            Node t = nm->mkNode(kind::SINGLETON, itmm.first);
+            Node t = nm->mkSingleton(itmm.first.getType(), itmm.first);
```

Comment:
You should use the element type of the type of eqc here. Otherwise this may break if a value is 1.0.

Reply:
Done. It is nice here we don't have to sets of different types in the same class.